test/integration/src/test/java/io/pravega/test/integration/KeyValueTableTest.java

```diff
@@ -192,6 +261,1427 @@ public void testCreateDeleteKeyValueTable() {
+        assertEquals("Corresponding key for entry2 should be as inserted",
+                key, tableEntry2.getKey().getKey());
+        assertEquals("Corresponding value for entry2 should be as inserted",
+                value2, tableEntry2.getValue());
+        assertNotEquals("Corresponding value for entry2 should not remian as entry1",
+                value1, tableEntry2.getValue());
+        assertNotEquals("Version1 and Version2 should be different in values", version1, version2);
+
+        // Updating third set of values in the table using version of second entry as condition.
+        String value3 = getValue();
+        Version version3 = newKeyValueTable.replace(null, key, value3, version2).join();
+        assertNotNull("Vesrion3 should not be null", version3);
+        log.info("version3: {}", version3);
+        TableEntry<Integer, String> tableEntry3 = newKeyValueTable.get(null, key).join();
+        log.info("Get tableEntry3: {}", tableEntry3);
+        assertEquals("Corresponding key for entry3 should be as inserted",
+                key, tableEntry3.getKey().getKey());
+        assertEquals("Corresponding value for entry3 should be as inserted",
+                value3, tableEntry3.getValue());
+        assertNotEquals("Corresponding value for entry3 should not remian as entry2",
+                value2, tableEntry3.getValue());
+        assertNotEquals("Version2 and Version3 should be different in values",
+                version2, version3);
+
+        // Trying to update fourth set of values in the table using version of first entry as condition.
+        String value4 = getValue();
+        AssertExtensions.assertSuppliedFutureThrows(
+                "Should not allow updating the table entry with an older key version",
+                () -> newKeyValueTable.replace(null, key, value4, version1),
+                ex -> ex instanceof BadKeyVersionException
+        );
+        log.info("Found BadKeyVersionException while updating the table entry with an older key version");
+
+        TableEntry<Integer, String> tableEntry4 = newKeyValueTable.get(null, key).join();
+        log.info("Get tableEntry4: {}", tableEntry4);
+        assertEquals("Corresponding key for entry4 should be as inserted",
+                key, tableEntry4.getKey().getKey());
+        assertNotEquals("Corresponding value4 for entry4 should not be as inserted",
+                value4, tableEntry4.getValue());
+        assertEquals("Corresponding value3 for entry4 should be as inserted",
+                value3, tableEntry4.getValue());
+        assertNotEquals("Corresponding value for entry4 should not remian as entry1",
+                value1, tableEntry4.getValue());
+        log.info("Successfully completed test Conditional update Key value using its version");
+    }
+
+    /**
+     * Test case-45: Conditional delete single TableEntry using its version
+     */
+    @Test
+    public void testConditionDeleteKey() {
```

Comment:
This is already covered in the base class.

Reply:
Sure, I'll remove this test as its a duplicate.